test/functional/integreatly_test.go

```diff
@@ -25,35 +26,15 @@ var _ = Describe("integreatly", func() {
 		}
 	})
 
+	rhmi, err := common.GetRHMI(k8sClient, true)
+	if err != nil {
+		t.Fatalf("error getting RHMI CR: %v", err)
+	}
+
 	RunTests := func() {
 
 		// get all automated tests
-		tests := []common.Tests{
-			{
-				Type:      fmt.Sprintf("%s ALL TESTS", installType),
-				TestCases: common.GetAllTestCases(installType),
-			},
-			{
-				Type:      fmt.Sprintf("%s HAPPY PATH", installType),
-				TestCases: common.GetHappyPathTestCases(installType),
-			},
-			{
-				Type:      fmt.Sprintf("%s IDP BASED", installType),
-				TestCases: common.GetIDPBasedTestCases(installType),
-			},
-			{
-				Type:      fmt.Sprintf("%s Functional", installType),
-				TestCases: FUNCTIONAL_TESTS,
-			},
-			{
-				Type:      fmt.Sprintf("%s SCALABILITY TESTS", installType),
-				TestCases: common.GetScalabilityTestCases(installType),
-			},
-			{
-				Type:      "FAILURE TESTS",
-				TestCases: common.FAILURE_TESTS,
-			},
-		}
+		tests := getTestsBasedOnInstallType(*rhmi)
```

Comment:
I know why this is being done but I don't like it. We are all the time moving away from the features that go test can provide us and instead we create our own logic. There is not ask with this comment only voicing a view.

Reply:
I agree, initial commit of this PR had it in a separate test suite, but then I've realized that it will take too much effort when it comes to interesting it with pipelines, generating report via delorean etc.